cpp/src/arrow/testing/random.cc

```diff
@@ -369,12 +371,16 @@ std::shared_ptr<Array> RandomArrayGenerator::FixedSizeBinary(int64_t size,
                                                 std::move(null_bitmap), null_count);
 }
 
-std::shared_ptr<Array> RandomArrayGenerator::Offsets(int64_t size, int32_t first_offset,
-                                                     int32_t last_offset,
-                                                     double null_probability,
-                                                     bool force_empty_nulls) {
-  using GenOpt = GenerateOptions<int32_t, std::uniform_int_distribution<int32_t>>;
-  GenOpt options(seed(), first_offset, last_offset, null_probability);
+namespace {
+template <typename ArrayType>
```

Comment:
When I read this first, I was looking for `typename ArrayType::TypeClass::offset_type` because I assumed this parameter referred to `StringArray` or so
```suggestion
template <typename OffsetArrayType>
```